### ISSUE TITLE
[subplat] [emails] Unprocessed payment email received after updating 3D secure card information

### ISSUE THREAD

**Issue by data-sync-user** [mozilla/fxa#10257]
**Affected versions**:

* All;

**Affected FxA versions**:

* 1.213.5;

**Prerequisites**:

* Have the VPN set to Italy;

**Steps to reproduce**:
1. Subscribe to Mozilla VPN monthly plan using a 3D Secure card: 4000002500003155;
2. Open your account email inbox and check the payment confirmation email;
3. Go to Subscription Management page for the same account;
4. Click on the “Change” button from Payment information section and update your card number with the following 3D secure card: 4000002760003184;
5. Wait for the next billing cycle and check your email for the same account;

**Expected result**:

* A mail with the title “Mozilla VPN Payment Received”  is visible in the inbox. The content of the email reflects the new used card.

**Actual result**:

* The “Mozilla VPN Payment Received” mail is not visible in the inbox. A payment failed mail is received instead. 

**Notes**:

* Reproducible only with 3D Secure cards.

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/FXA-3907)
┆Attachments: <a href="https://mozilla-hub.atlassian.net/secure/attachment/16136/image-20210826-105858.png">image-20210826-105858.png</a> | <a href="https://mozilla-hub.atlassian.net/secure/attachment/16135/image-20210826-105930.png">image-20210826-105930.png</a>

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

Joseph Anthony Cuevas Do you know if Support can update Stripe CCs? Just curious if there are workarounds if a user runs into this, though hopefully we can investigate soon.

**Comment by data-sync-user** [comment]
➤ Joseph Anthony Cuevas commented:

Bianca Danforth No we do not have the ability to do this in Stripe. Typically we instruct customers to log in to their subscription and change it themselves. At worst we will cancel and have them resub with the new card.

**Comment by data-sync-user** [comment]
➤ Barry Chen commented:

Madalin Cotetiu  the 3184 card requires SCA on every transaction, which wouldn’t work for recurring charges on our subscription right now.  Can you re-test by using 4000003800000446 as the new card?  Thanks!

**Comment by madalincm** [comment]
Sure! I made a subscription using the following 3D Secure card: 4000002500003155. I updated the card info with the following card: 4000003800000446 and now I'm waiting for the recurrency email.

**Comment by madalincm** [comment]
I received the recurrency email and the content of the email reflects the new used card. See the attached screenshots.

![Screenshot 2021-09-08 at 12 51 29](https://user-images.githubusercontent.com/15685441/132674366-b731eecb-5e0d-4e18-9341-d44482c7ca29.png)
<img width="648" alt="Screenshot 2021-09-08 at 12 52 26" src="https://user-images.githubusercontent.com/15685441/132674381-0326e496-6a8a-46e0-8eda-2ff6659bf595.png">
<img width="881" alt="Screenshot 2021-09-08 at 12 53 13" src="https://user-images.githubusercontent.com/15685441/132674394-d97ca19d-a3fd-4199-9c6c-f5b26fa00105.png">
![Screenshot 2021-09-09 at 13 57 37](https://user-images.githubusercontent.com/15685441/132674435-aa6d613c-9475-49e9-bf66-acecd030ec33.png)

**Comment by data-sync-user** [comment]
➤ Barry Chen commented:

Madalin Cotetiu  Thank you for the verification.

Bianca Danforth I see that you’re the assignee.  Is there anything to do for this jira?  I think we can cancel it?

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

Unfortunately, we have never supported “off-session” authentication for +SCA+ ( https://stripe.com/docs/payments/3d-secure ). By that I mean we don’t distinguish between a true failed payment (e.g. due to insufficient funds) and a charge that requires SCA before it can be captured. Both cases receive our payment failed email.

So this leads me to ask: Is it possible that a subsequent charge will require SCA? The short answer is yes. Generally, a fixed-amount subscription and merchant-initiated transaction are considered exceptions and SCA would not be required for our recurring charges, but it is possible for the bank to require it anyway ( https://stripe.com/docs/strong-customer-authentication/migration#recurring ). Because of this, Stripe recommends having a way for users to authenticate off-session, and provides APIs that can tell us if capturing a charge failed because SCA is required ( https://stripe.com/docs/billing/migration/strong-customer-authentication#scenario-3 ).

I have done some checks, and I think we can support “off-session” authentication for SCA by:

  * Modifying the copy of the failed payment email slightly to indicate failure could be due to a necessary authentication step.
  * Updating the customer from the auth server in payments with an additional error property (something similar to FXA-3088).
  * In Subscription Management, check for this new error property and display a descriptive message in an AlertBar (similar to FXA-2806).
  * The user can then enter the same card as before and authenticate as part of submitting the card information in the PaymentUpdateForm.

I don’t actually know how many users are affected by this, as my suspicion is that the vast majority of CC customers for which SCA is required only need to authenticate for the initial payment. If possible, I’d like to check to see how many subsequent charges have required SCA historically. I’ll try to do that next week.

**Comment by data-sync-user** [comment]
➤ Barry Chen commented:

Bianca Danforth Thanks for the research and detailed notes.  That is certainly something we could do, but I don’t think it’s small or maybe even medium.  That’s partly because I don’t think we should have a single vague email for all payment failures.  If we know it’s SCA we should tell the customer what they need to do to fix it.

I do like the idea.  Let’s go for it?

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

Using this temporary Node script ( https://gist.github.com/biancadanforth/5e6e2d9cfa0f206e3d1c1a02f7778b5e ) I wrote, I checked to see how many subscriptions have required “off-session” SCA in the past 30 days in prod. There were 0 recurring invoices and therefore 0 affected subscriptions. All 8 of the invoices found were the first invoice only i.e. “on-session” SCA, which we do support.

Consequently, since no users are affected by this, I am going to mark this as WONTFIX for now.
